src/api/dirmonitor/polling.c

```diff
@@ -0,0 +1,126 @@
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <time.h>
+#include <SDL.h>
+
+#define EVENT_MODIFIED (1 << 0)
+
+struct file_info {
+  int fd;
+  time_t mtime;
```

Comment:
is there other stuff we may need to include?
How about st_dev and st_ino to track if the folder changes completely?

Reply:
I am okay with that. Sounds reasonable enough.